processor/src/chiplets/hasher/mod.rs

```diff
@@ -25,6 +25,16 @@ pub use aux_trace::{AuxTraceBuilder, SiblingTableRow, SiblingTableUpdate};
 #[cfg(test)]
 mod tests;
 
+/// Result of a merkle tree node update. The result contains the old merkle_root, which
+/// corresponding to the old_value, and the new merkle_root, for the updated value. As well as the
+/// row address of the execution trace at which the computation started.
+#[derive(Debug, Copy, Clone)]
+pub struct MerkleRootUpdate {
+    pub address: Felt,
```

Comment:
optional nit: exposing the types as `pub` will remove any type-safety we might want to achieve as the user can just fetch a merkle update, change one of the roots and propagate the type. OTOH, if we only have protected attributes, whenever we have an instance of this type, we have the guarantee it was properly generated by the exposed API of this library.

If we opt to keep the attributes as `pub` for the sake of simplicity, it is a good thing to add documentation individually to each of the attributes - as it would be if we had individual getters for them.

Reply:
I was on the fence about this too. I generally prefer not to add the `pub`, though it does feel a bit heavy to add getters. I'd probably lean towards it though.

Reply:
I'm not sure if this is better in this case, without the fields being public de-structuring is not possible. So this codes get a bit more complicated:

https://github.com/0xPolygonMiden/miden-vm/pull/652/files#diff-df7a28227843c639908744c2846d55bf5dab202e016cdd4bcab063b587e5140aR150-R154

We also move the same problem to the `new` method (although it is better to have this is a single place instead of multiple)

It doesn't really prevent error mentioned, because `MerkleRootUpdate::new` would still be available (the alternative would be to not move this to the super [crate](https://github.com/0xPolygonMiden/miden-vm/pull/652/files#r1084293759) or to use `path` visibility), and I think if a user thought it would be a good idea to change the fields via direct access, there is not much preventing the same user of creating a new instance. I guess if we really want to prevent such changes both the fields and new should be private and to the hasher module, to prevent both changes and new instances from being created.

Reply:
Let me know if I should do the change :)

Reply:
Done 1. fields are no longer public 2. moved to the parent module 3. added the header